src/ValidatorInterface.php

```diff
@@ -13,8 +13,7 @@ interface ValidatorInterface
      * Validate data set against rules set for data set attributes.
      *
      * @param DataSetInterface|mixed|RulesProviderInterface $data Data set to validate.
-     * @param Rule[][] $rules Rules to apply.
-     * @psalm-param iterable<string, Rule[]> $rules
+     * @param iterable<RuleInterface|RuleInterface[]> $rules Rules to apply.
```

Comment:
```suggestion
     * @psalm-param iterable<RuleInterface|RuleInterface[]> $rules Rules to apply.
```

Reply:
Why?

Reply:
`iterable<RuleInterface|RuleInterface[]>` is psalm-specific, is that right?

Reply:
Scrutinizer doesn't recognize generics in non-psalm annotations.

Reply:
<img width="841" alt="image" src="https://user-images.githubusercontent.com/6815714/164728972-98b33d91-117b-4fe3-8a00-8c99749329dd.png">

Both psalm and scrutinizer are work.
PHPStorm correctly recognises type of array.

<img width="739" alt="image" src="https://user-images.githubusercontent.com/6815714/164729447-8bc8d689-afe9-4e07-a591-1c3faac3023e.png">